### PR TITLE
chore: run migrations conditionally

### LINKED ISSUE #13921: [CAL-3243] Run auto migrations conditionally
* If DIRECT_DATABASE_URL is not set, skip migrations
* If it's set but fails. It should prevent deployment from continuing.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3243](https://linear.app/calcom/issue/CAL-3243/run-auto-migrations-conditionally)</sub>

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Fixes #13921

- If DIRECT_DATABASE_URL is not set, skip migrations
- If it's set but fails. It should prevent deployment from continuing.

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- yarn build with DATABASE_URL undefined (for DB-less builds)
- yarn build with DATABASE_DIRECT_URL undefined
- yarn build with DATABASE_DIRECT_URL defined
- yarn build with a failed migration
- yarn build withouth a failed migration

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/prisma/auto-migrations.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * TODO: re-write this when Prisma.io gets a programmatic migration API
+ * Thanks to @olalonde for the idea.
+ * @see https://github.com/prisma/prisma/issues/4703#issuecomment-1447354363
+ */
+async function main(): Promise<void> {
+  if (!process.env.DATABASE_DIRECT_URL) {
+    console.info("No DATABASE_DIRECT_URL found, skipping migration");
+    return;
+  }
```
Comment (zomars) [inline packages/prisma/auto-migrations.ts:18]: Working on Vercel 
image

File: packages/prisma/auto-migrations.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import dotEnv from "dotenv";
+import { exec as execCb } from "node:child_process";
+import { promisify } from "node:util";
+
+dotEnv.config({ path: "../../.env" });
```
Comment (zomars) [inline packages/prisma/auto-migrations.ts:5]: Load env locally.

File: packages/prisma/auto-migrations.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    env: {
+      ...process.env,
+    },
+  });
+  console.log(stdout);
+}
+
+main().catch((e) => {
+  console.error(e.stdout);
+  process.exit(1);
```
Comment (zomars) [inline packages/prisma/auto-migrations.ts:30]: This will prevent the build from continuing on error

**Review by zomars — COMMENTED** [review]
Self review done

**Review by keithwillcode — DISMISSED** [review]

**Review by keithwillcode — DISMISSED** [review]

**Review by keithwillcode — APPROVED** [review]

**Review by keithwillcode — DISMISSED** [review]
Production build is consistently failing

**Review by emrysal — APPROVED** [review]

**Pull request merged**